### PR TITLE
CHK-425: Extract shipping list and option components to entrypoints

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Adds the `ShippingOptionList`, `ShippingOption`, `ShippingHeader` and `useAddressRules` components to be used by other apps (such as `shipping-calculator`).

#### How should this be manually tested?

[Workspace](https://lucas--checkoutio.myvtex.com/cart/add?sku=289).

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Review by jeffersontuc — APPROVED** [review]
LGTM 🚀

File: react/ShippingHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  onEditAddress?: () => void
+}
+
+const { useAddressContext } = AddressContext
+
+const ShippingHeader: React.VFC<Props> = ({ onEditAddress }) => {
+  const { address, invalidFields } = useAddressContext()
+
+  if (!address || invalidFields.includes('postalCode')) {
+    return <p className="t-body mt0 mb6">Por favor informe seu endereço</p>
```
Comment (wdsrocha) [inline react/ShippingHeader.tsx]: suggestion: use `FormattedMessage`

File: react/ShippingOptionList.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    showSelectedDeliveryOption ? (
+      <>
+        <p className="mt0 mb5 fw4 f4 lh-copy">
+          <FormattedMessage id="store/checkout.shipping.selectedDeliveryOptionsLabel" />
+        </p>
+
+        <ListGroup>
+          {selectedDeliveryOptions.map((deliveryOption) => (
+            <ShippingOption
+              key={deliveryOption.id}
```
Comment (wdsrocha) [inline react/ShippingOptionList.tsx:67]: question: not sure if that matters here, but why not using the non-null assertion here as well? Same for line 83.

File: react/ShippingHeader.tsx (hunk 1 above)
Comment (lucasecdb) [reply react/ShippingHeader.tsx]: nice catch!

File: react/ShippingOptionList.tsx (hunk 2 above)
Comment (lucasecdb) [reply react/ShippingOptionList.tsx:67]: because the types for a component key allows null values as well

File: react/ShippingForm.tsx [hunk 3]
```diff
@@ -101,13 +105,31 @@ const ShippingForm: React.FC = () => {
...
+                    <IconEdit solid />
+                  </ButtonPlain>
+                </div>
+              </span>
+
+              <div className="mt2 mb6 lh-copy">
+                {selectedAddress.receiverName}
+              </div>
+            </div>
+          )}
```
Comment (wdsrocha) [inline react/ShippingForm.tsx:124]: question: how to be sure that this will never be triggered on shipping calculator?

Comment (wdsrocha) [reply react/ShippingForm.tsx:124]: it seems that shipping won't use the state machine, is that right, @lucasecdb?

Comment (lucasecdb) [reply react/ShippingForm.tsx:124]: it won't for now, but we'd like to make it use it at some point in the future (at least a subset of the states)

File: react/ShippingOptionList.tsx (hunk 2 above)
Comment (wdsrocha) [reply react/ShippingOptionList.tsx:67]: sure, but I just wondered if it would be good to raise a warning in this case, as it isn't an expected behavior

**Review by wdsrocha — APPROVED** [review]

File: react/ShippingOptionList.tsx (hunk 2 above)
Comment (lucasecdb) [reply react/ShippingOptionList.tsx:67]: I don' think the delivery option id can _actually_ be `null`, maybe the typings on our GraphQL are incorrect

**Pull request merged**